node/node.go

```diff
@@ -376,13 +376,24 @@ func (n *Node) obtainJWTSecret(cliParam string) ([]byte, error) {
 // startup. It's not meant to be called at any time afterwards as it makes certain
 // assumptions about the state of the node.
 func (n *Node) startRPC() error {
-	if err := n.startInProc(); err != nil {
+	// Filter out personal api
+	apis := n.rpcAPIs
+	for i, api := range apis {
+		if api.Namespace == "personal" {
+			if !n.config.EnablePersonal {
+				apis = append(apis[:i], apis[i+1:]...)
```

Comment:
Deleting like this is not compatible with range looping over the same set of elements. It ends up in very very weird things: https://go.dev/play/p/yeiziOFupG2

You want the outer loop to be with an iterator from `0` to `len(apis)`, and decrement `i` when an item is deleted.

In this specific face the end result might accidentally work because it's only ever a single item being deleted, but the execution itself is a horrible mess that accidentally produces the correct result.